include/dynd/type.hpp

```diff
@@ -281,13 +281,99 @@ namespace ndt {
       return match(other, tp_vars);
     }
 
+    template<typename T>
+    bool has_scalar_type() const
+    {
+      switch (get_id()) {
+      case bool_id: return std::is_same<T, type_of<bool_id>::type>::value;
+      case int8_id: return std::is_same<T, type_of<int8_id>::type>::value;
+      case int16_id: return std::is_same<T, type_of<int16_id>::type>::value;
+      case int32_id: return std::is_same<T, type_of<int32_id>::type>::value;
+      case int64_id: return std::is_same<T, type_of<int64_id>::type>::value;
+      case int128_id: return std::is_same<T, type_of<int128_id>::type>::value;
+      case uint8_id: return std::is_same<T, type_of<uint8_id>::type>::value;
+      case uint16_id: return std::is_same<T, type_of<uint16_id>::type>::value;
+      case uint32_id: return std::is_same<T, type_of<uint32_id>::type>::value;
+      case uint64_id: return std::is_same<T, type_of<uint64_id>::type>::value;
+      case uint128_id: return std::is_same<T, type_of<uint128_id>::type>::value;
+      case float16_id: return std::is_same<T, type_of<float16_id>::type>::value;
+      case float32_id: return std::is_same<T, type_of<float32_id>::type>::value;
+      case float64_id: return std::is_same<T, type_of<float64_id>::type>::value;
+      case float128_id: return std::is_same<T, type_of<float128_id>::type>::value;
+      case complex_float32_id: return std::is_same<T, type_of<complex_float32_id>::type>::value;
+      case complex_float64_id: return std::is_same<T, type_of<complex_float64_id>::type>::value;
+      default: return false;
+      }
+    }
+
+    template<typename T>
+    bool has_admissible_value_type() const
+    {
+      switch (get_id()) {
+      case string_id: return std::is_same<T, std::string>::value;
+      case type_id: return std::is_same<T, ndt::type>::value;
+      default: return false;
+      }
+    }
+
+    template<typename T>
+    bool has_admissible_property_type() const
+    {
+       return has_scalar_type<T>() || has_admissible_value_type<T>();
+    }
+
     /**
```

Comment:
This comment should probably be moved down to where the property access method is.